### PR TITLE
Fix periodic resample

### PR DESCRIPTION
**Pull request by vchigrin**

### CODE REVIEW

**Comment by vchigrin** [comment]
Just signed individual CLA.

File: tensorflow/contrib/periodic_resample/BUILD [hunk 1]
```diff
@@ -69,6 +70,17 @@ py_library(
     ],
 )
 
+py_test(
```
Comment (vchigrin) [inline tensorflow/contrib/periodic_resample/BUILD]: Not sure whether this change is correct. I have added it to run this test by  "bazel test " on Linux. If this is mistake and it is possible to  run test without this, please, tell me about proper way doing this.

**Comment by jhetherly** [comment]
@vchigrin I'm just catching up on this thread. Thanks for adding the check for the desired shape and fixing the MathJax (seems I forgot the double slash). Regarding the documentation, I notice that most of my indentation is ignored. Do you know how to remedy this so that the formatting in the doc string is more-or-less respected on the documentation webpage?

**Comment by vchigrin** [comment]
@jhetherly  - Sorry, I am very novice in Tensorflow, so I also don't know much about this. Will search the repo, if I find example code, will fix it too.

By the way, I locally implemented gradient propagation in this operation, and made output shape fully defined, when possible. May be it will be interesting to you to look at these changes. I can make PR with them if it worth it. See last 2 commits in 
https://github.com/vchigrin/tensorflow/commits/periodic-resample-grad

**Comment by jhetherly** [comment]
@vchigrin Ah, I'll also hunt for ways to make the docs formatted a bit better. As for the gradient operations, it seems like you refactored some of the common components for the op and essentially did identical operations for both the op and the grad. If this summary is correct, then I greatly appreciate this addition. This was next on my todo list, but I hadn't found the time. Thanks a lot!

**Comment by vchigrin** [comment]
@jhetherly Seems I've found way to  fix line breaks - it's neccessary to use two newlines instead of one. Tried to build documentation from latest commit, then used  pandoc to generate HTML from markdown files. Here is how generated HTML looks like now, please, check is this  something you expect?
https://jsfiddle.net/tLdougwr/
By  the way, please, check whether last expression for  hq is  correct, I'm a bit confused -  Sq present in it twice....

About my changes with  gradient propagation - yes, you are absolutely correct - I tried touse existing code as  much as possible. Index computation is tricky and I was afraid to make some hard to debug errors...

**Comment by jhetherly** [comment]
@vchigrin I now see that it's a bit unclear, but the first $S_q$ is capitalized and is define earlier while the second $s_q$ is lower case and represents the index of the $q$th component of the output tensor. One small typo is where I give the definition of $S_q$ I wrote "defined as by." Could you remove the "as" in that sentence? Thanks again for this PR!

**Comment by vchigrin** [comment]
@jhetherly , Fixed it. Thanks for explaining that equation!

**Comment by vchigrin** [comment]
@rmlarsen - Could you review this PR, please?

**Comment by vchigrin** [comment]
@rmlarsen , @drpngx : Already after creation of this PR I made a bunch of even more useful changes in this operation in my local repository:

- Added support for back propagating gradients for this operation.
- Optimized it to allow compute indices incrementally - that speed up it 2.5 times (single-thread version).
- Made possible split this operation across mutliple threads - that make it work even faster on modern multi-core CPU.

Since this review, unfortunately, is not approved yet, I have a question: What is the better way of contributing my improvements? Should I push them all in this PR so all of them will be reviewed simultaneously, or it is better to wait for this PR to be approved and then make another one?

**Comment by vchigrin** [comment]
@rmlarsen , @drpngx - could anybody look at this PR? This is my first PR to Tensorflow, if I missed something I need to get these changes reviewed, please, tell me. I will be happy to contribute 
my improvements to the project.

**Comment by drpngx** [comment]
@jhetherly WDYT?

**Comment by jhetherly** [comment]
@drpngx : From my conversation @vchigrin and reviewing the changes to the code he's made, the changes look quite beneficial.
@vchigrin, just to be sure, do the python tests in still pass with your changes (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/periodic_resample/python/kernel_tests/periodic_resample_op_test.py)? If so, I think we're good to go. If it's possible, it may be nice to include additional python tests for the gradients since you added that info into the op.

**Review by rmlarsen — APPROVED** [review]

**Comment by rmlarsen** [comment]
@vchigrin Thanks for the change.
@jhetherly Thanks for reviewing and helping improve it.

**Comment by jhetherly** [comment]
@vchigrin: unless I missed it, this merge doesn't seem to include your contributions of the gradient info, etc. Do you plan to create another PR for these (__very__ useful) additions?

**Comment by vchigrin** [comment]
@jhetherly : Sorry for delay - TensorFlow is my hobby, not primary work area. Yes, I will upstream these changes too, hope on this weekand. About your earlier questoions -  this PR does not have changes for gradient, that's why there are no new tests for it. New PR will have them  - I already wrote it. As far as I know all tests for this operation passed successfully on my machine. 

By the way, is it correct that infrastructure in this repository will not allow to merge PR with broken tests?  I am very new in TensorFlow development... but as far as I know some projects implement such policy...

**Comment by jhetherly** [comment]
@vchigrin: Ah, I understand (no need for an apology ;) ).
I'm not sure what the merge policy is for TF.

**Comment by vchigrin** [comment]
@jhetherly : Made PR https://github.com/tensorflow/tensorflow/pull/16520 - could you look at it? I extracted most index computation logic to separate class, this made implementing intra-op parallelism much easier.

I tested it locally by command
bazel test --config=opt --test_output=all '//tensorflow/contrib/periodic_resample/...'
all tests passed successfully.

**Pull request merged**